File: 0101_0200/0170/0170.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.read())
    i = int(((n * 8 - 3) ** 0.5 - 1) // 2 + 1)
    while i > 1 and not (n > (i - 1) * i / 2 and (n - ((i - 1) * i / 2)) % i == 0):
        i -= 1
    q.write(str(i))

File: 0101_0200/0170/0170.go
package main

import (
	"fmt"
	"math"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n int
	fmt.Fscan(f, &n)
	i := (int(math.Sqrt(float64(n*8-3)))-1)/2 + 1
	for ; i > 1 && !(n > (i-1)*i/2 && (n-((i-1)*i/2))%i == 0); i-- {
	}
	fmt.Fprint(q, i)
}
